packages/fxa-auth-db-mysql/lib/db/mem.js

```diff
@@ -1086,6 +1086,33 @@ module.exports = function (log, error) {
     }).reverse())
   }
 
+  Memory.prototype.securityEventsByUid = function (uid) {
+    if (! uid) {
+      return P.reject(error.notFound())
+    }
+    var key = uid.toString('hex')
+    var events = securityEvents[key] || []
+    var securityEventsLimit = 50
+
+    var filteredSecurityEvents = events.filter(function (ev) {
+      return ev.uid.toString('hex') === key
+    }).reverse()
+
+    if (filteredSecurityEvents.length > securityEventsLimit) {
```

Comment:
This will update the `length` property of filteredSecurityEvents but won't actually limit the array size to 50. 

An alternative is to use 2 filters to ensure there are only 50 items in the list. Look up the the function signature of events.filter, the callback takes multiple arguments, the 2nd is an index.